go/cmd/vtgate/plugin_promstats.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+// This plugin imports Prometheus to allow for instrumentation
+// with the Prometheus client library
+
+import (
+	_ "github.com/youtube/vitess/go/stats/promstats"
```

Comment:
The plugin should also initialize the prefix used in the metric registration options -- e.g. "vtgate" in this case should be the "Namespace" in the prom options:

https://github.com/prometheus/client_golang/blob/master/prometheus/metric.go#L68

This is going to be a bit tricky, since even if you do  it inside an `init()` block, it won't necessarily run before all the other static initializers run.

Thinking more about this, what I think we need to do here (unfortunately) is keep enough info in the `stats` object (e.g. the snake_name and the help string) to be able to lazily initialize the prom exporter after stats are registered.